src/Domain/CoordinateMaps/CubicScale.cpp

```diff
@@ -101,168 +168,258 @@ boost::optional<std::array<tt::remove_cvref_wrap_t<T>, 1>> CubicScale::inverse(
   // version, here we utilize the boost implementation, as it includes
   // additional checks for zero derivative, checks on bounds, and can implement
   // bisection if necessary.
-  return {
-      {{outer_boundary_ * RootFinder::newton_raphson(
-                              cubic_and_deriv, initial_guess, 0.0, 1.0, 14)}}};
+  const double scale_factor =
+      RootFinder::newton_raphson(cubic_and_deriv, initial_guess, 0.0, 1.0, 14) /
+      target_dimensionless_radius;
+
+  std::array<tt::remove_cvref_wrap_t<T>, Dim> result{};
```

Comment:
Why not initialize to scale_factor * target_coords[0] and then loop from i = 1 to i<Dim? This kind of optimization was done above

Reply:
Sorry, I'm not sure what you're referring to

Reply:
It's not a big deal. I just wondered why you don't do something like

~~~
std::array<tt::remove_cvref_wrap_t<T>, Dim> result{square(dereference_wrapper(source_coords[0]))};
for (size_t i = 1; i < Dim; ++i) {
// ...
}
~~~

as you did above, instead of initializing `result{}` and then looping from 0 instead of from 1.

Reply:
Ah, I think I understand now. You mean the line:
```cpp
    boost::optional<std::array<tt::remove_cvref_wrap_t<T>, Dim>> result{
        std::array<tt::remove_cvref_wrap_t<T>, Dim>{}};
```
This is done to make the `boost::optional` have a default-constructed `std::array` because the default-constructed `boost::optional` doesn't have a valid value. I'll add a comment as I squash :)